rest_framework/fields.py

```diff
@@ -674,9 +675,11 @@ class BooleanField(Field):
         '0', 0, 0.0,
         False
     }
+    NULL_VALUES = {'n', 'N', 'null', 'Null', 'NULL', '', None}
 
     def __init__(self, **kwargs):
-        assert 'allow_null' not in kwargs, '`allow_null` is not a valid option. Use `NullBooleanField` instead.'
+        if django.VERSION < (2, 1):
+            assert 'allow_null' not in kwargs, '`allow_null` is not a valid option. Use `NullBooleanField` instead.'
```

Comment:
We can actually support this for all Django versions for the serializer field, it's just that the *model* would need to use a `models.NullBooleanField`.

Ie. I think we can just drop the check completely.